Update documentation for CIP on consistency rewards (#603)
cowprotocol/docs: docs/cow-protocol/reference/core/auctions/rewards.md

# Solver rewards

> The protocol is currently subsidizing the solver competition on all chains it operates on, by rewarding solvers on a weekly basis (currently, every Tuesday) with rewards paid in COW. Solvers are rewarded based on their performance as solvers (i.e., when participating in the standard solver competition) as specified by [CIP-20](https://snapshot.org/#/cow.eth/proposal/0x2d3f9bd1ea72dca84b03e97dda3efc1f4a42a772c54bd2037e8b62e7d09a491f), [CIP-36](https://snapshot.org/#/cow.eth/proposal/0x4e58f9c1208121c0e06282b5541b458bc8c8b76090263e25448848f3194df986), [CIP-38](https://snapshot.org/#/cow.eth/proposal/0xfb81daea9be89f4f1c251d53fd9d1481129b97c6f38caaddc42af7f3ce5a52ec), [CIP-48](https://snapshot.org/#/cow.eth/proposal/0x563ab9a66265ad72c47a8e55f620f927685dd07d4d49f6d1812905c683f05805) [CIP-57](https://snapshot.box/#/s:cow.eth/proposal/0x46d4fea1492207cf400fcb7a01141a7d4c730791d658cc77236941fc9eb7dccb), [CIP-67](https://snapshot.box/#/s:cow.eth/proposal/0xf9ecb08c4738f04c4525373d6b78085d16635f86adacd1b8ea77b2c176c99d32), [CIP-74](https://snapshot.org/#/s:cow.eth/proposal/0x0c70c8cd92accee872b52614b4fa10e3e3214f45c5b6857f7e88e910607a3c1d), and CIP-XX(add snapshot link). Solver rewards for participating in the price estimation competition and providing quotes that are needed for the gas estimates and limit price computations of market orders are specified in [CIP-27](https://snapshot.org/#/cow.eth/proposal/0x64e061568e86e8d2eec344d4a892e4126172b992cabe59a0b24c51c4c7e6cc33), [CIP-36](https://snapshot.org/#/cow.eth/proposal/0x4e58f9c1208121c0e06282b5541b458bc8c8b76090263e25448848f3194df986), [CIP-57](https://snapshot.box/#/s:cow.eth/proposal/0x46d4fea1492207cf400fcb7a01141a7d4c730791d658cc77236941fc9eb7dccb), and [CIP-72](https://snapshot.box/#/s:cow.eth/proposal/0xc1b1252f0c99126b4e09730022faa31a7bb58073a3dc064c19b74d44164c39a7).

Reviewer: The terms used are a bit confusing, in my opinion. You write

>  Solvers are rewarded based on their performance as solvers (i.e., when participating in the standard solver competition) 

This already mentions performance, but in the explanatory parenthesis, participation shows up.

Then in the next section "Solver competition rewards ", we start with a subsection "Performance rewards".

I would suggest we explicitly define the terms and make the split between performance and consistency to make

:::note

For the interested reader, the main source of truth for the weekly payments to solvers is this [Dune dashboard](https://dune.com/cowprotocol/cow-solver-rewards). The dashboard is populated with data aggregated by scripts within the [solver-rewards](https://github.com/cowprotocol/solver-rewards) repository.

:::

## Solver competition rewards (CIPs 20, 36, 38, 48, 57, 67, 74, XX)

> ### Performance rewards

Reviewer: The first sentence in this section starts with "solver rewards...." Probably we should use "Performance rewards...."

:::note

The performance reward calculation can result in a negative value, in which case the solver is required to pay the protocol.

:::

The performance reward is capped from above and below using the function $$\textrm{cap}(x) = \max(-c_l, \min(c_u, x))$$, where $$c_u$$ is equal to a chain-specific $$\beta$$ percent of the protocol fee (excluding partner fees) that the protocol earned from the trades in all solutions successfully executed onchain by the solver in that auction, and $$c_l$$ is chain-specific as well. $$\beta$$ and $$c_l$$ are determined by the following values:

- Ethereum mainnet, Arbitrum, and Base chain: $$\beta = 50\%$$, $$c_l = 0.010 \;\textrm{ETH}$$
- Gnosis Chain: $$\beta = 100\%$$, $$10 \;\textrm{xDAI}$$
- Avalanche: $$\beta = 100\%$$, $$0.3 \;\textrm{AVAX}$$
- Polygon: $$\beta = 100\%$$, $$30 \;\textrm{POL}$$
- BNB: $$\beta = 100\%$$, $$0.04 \;\textrm{BNB}$$
- Linea and Ink: $$\beta = 100\%$$, $$0.0015 \;\textrm{ETH}$$
- Plasma: $$\beta = 100\%$$, $$30 \;\textrm{XPL}$$

> The parameter $$\beta$$, which naturally corresponds to a revenue-sharing parameter between protocol and solvers, is set to 50% by default. The core team has a mandate to change this parameter for individual networks to a value in the interval [50%, 100%], given that the network has a total revenue less than 5% of the revenue on mainnet.

Reviewer: ```suggestion
The parameter $$\beta$$, which naturally corresponds to a revenue-sharing parameter between protocol and solvers, is set to 50% by default. The core team has a mandate to change this parameter for individual networks, if needed, to a value in the interval [50%, 100%], given that the network has a total revenue less than 5% of the total protocol revenue.
```

:::

The performance reward to the winning solver $$i$$ is

$$
\textrm{performanceReward}_i = \textrm{cap}(\textrm{totalScore} - \textrm{referenceScore}_i-\textrm{missingScore}_i).
$$

Here $$\textrm{totalScore}$$ is the sum of the scores of all winning solutions in the auction and $$\textrm{missingScore}_i$$ is the sum of the scores of solver $$i$$'s winning solutions that reverted. Finally, $$\textrm{referenceScore}_i$$ is the total score of a counterfactual auction in which all bids from solver $$i$$ are removed from the set of bids that survive the fairness filtering.

:::note

> The performance reward calculation can result in a negative figure, in which case the solver is required to pay the protocol.

Reviewer: i do find the use of the word "figure" here a bit awkward. I would just write "negative number" or "negative value"

:::

### Consistency rewards

> In each auction, solver $$i$$'s contribution to the consistency budget is

Reviewer: Maybe start with 1-2 sentences explaining what consistency rewards are? This reference to "consistency budget" looks a bit abrupt

$$
\frac{1}{2} \cdot \textrm{protocolFee}_i - \max(0, \textrm{performanceReward}_i),
$$

> where $$\textrm{protocolFee}_i$$ is the protocol fee (excluding partner fees) earned from the trades in all solutions supplied by solver $$i$$. When the performance reward is a penalty (i.e., negative), the full $$\frac{1}{2} \cdot \textrm{protocolFee}_i$$ is contributed to the consistency budget. The total consistency budget for an accounting period is the sum of these contributions across all solvers and auctions.

Reviewer: "...earned from the trades in all solutions supplied by...."

not a being fan of the phrasing here. I would be more explicit. Something like

"earned from the trades in all solutions that got successfully executed onchain by solver i"

> The consistency budget is distributed at the end of each accounting period according to a consistency metric. The core team has a mandate to adapt this metric over time; every change will be announced in advance on the [CoW Protocol forum](https://forum.cow.fi).

Reviewer: We don't have to keep adapting, so i would probably rephrase as 

"The core team has a mandate to adapt this metric when they see fit"

To summarize, truthfully revealing the (cost-adjusted) score that a solver can generate for each submitted solution is optimal if the cap is not binding, and misreporting does not affect $$\textrm{referenceScore}_i$$. It is not necessarily optimal in uncompetitive auctions when the difference between the best and second-best solution may be large, and in some edge cases in which a solver may benefit from making the filtering step less stringent. However, in these cases, deriving the optimal strategy is a very complex problem.

Consistency rewards introduce an additional strategic dimension: since the order-count metric rewards solvers for submitting solutions to auctions in which orders are ultimately executed, solvers have an incentive to participate broadly across auctions, even in cases where they do not expect to win the performance reward.

> ## Price estimation competition rewards (CIPs 27, 36, 57, 72)

Reviewer: @fhenneke can you check to see if there is an issue here?
